addons/test_mail/tests/test_mail_activity.py

```diff
@@ -917,6 +917,29 @@ def test_get_activity_data(self):
                 'user_assigned_ids': record_activities.user_id.ids,
             })
 
+    @freeze_time("2023-10-18 06:00:00")
+    @users('employee')
+    def test_search_and_group_on_activity_state(self):
```

Comment:
Can't it be added in "test_mail_activity_mixin_search_state_basic" to avoid yet another test + data setup ?

Reply:
Hello @tde-banana-odoo, 

Yes, that’s a good idea. I have moved the test there but actually it was not easy as the test was failing there because an inconsistency between the method “read_group_groupby” and “search_activity_state” in mail_activity_mixin. So I have also modified a bit “read_group_groupby” so that it uses the same computation as “search_activity_state”. What do you think ? (maybe more tests are needed for such change)

(runbot is red but apparently for known issue but I will rebase it later to make it green)